src/bliss/util/execute.h

```diff
@@ -12,33 +15,35 @@ typedef unsigned long value_type;
 namespace bliss {
 namespace utils {
 namespace executor {
+
 void execute_inserts(bliss::BlissIndex<key_type, value_type> &tree,
-                     std::vector<key_type>::iterator &start,
-                     std::vector<key_type>::iterator &end, int seed = 0) {
+                     std::vector<key_type>::iterator start,
+                     std::vector<key_type>::iterator end, int seed = 0) {
     spdlog::trace("Executing Inserts");
     std::mt19937 gen(seed);
-    std::uniform_int_distribution<value_type> dist(0, 1);
+    auto num_keys = std::distance(start, end);
+    std::uniform_int_distribution<value_type> dist(0, num_keys - 1);
```

Comment:
Probably change this from `value_type` to `size_t` as `value_type` is not guaranteed to be a numeric type.

Reply:
Just updated!